### PR TITLE
fetcher.is_matching() should work even when match is None

### PR DESCRIPTION
**Pull request by fspot**
## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -20,7 +20,7 @@ lint:
 
 .PHONY: mypy
 mypy:
-	mypy peakina
+	mypy .
```
Comment (PrettyWood) [inline Makefile:23]: I think it's ok to enforce mypy only on the source code and not the tests. Why this change?

File: setup.cfg [hunk 2]
```diff
@@ -68,8 +68,11 @@ force_grid_wrap=0
 combine_as_imports=True
 
 [mypy]
-ignore_missing_imports = True
 plugins = pydantic.mypy
+follow_imports = silent
+ignore_missing_imports = True
+allow_redefinition = True
+#check_untyped_defs = True
```
Comment (PrettyWood) [inline setup.cfg:75]: I also want mypy to be stricter but I think it should not be done in a fix. While you are it, can be try with `strict = True` ?

Comment (fspot) [reply setup.cfg:75]: If mypy enabled on `peakina` only:
- 0 errors with check_untyped_defs
- 106 errors with strict

If mypy enabled on `.`:
- 26 errors with check_untyped_defs
- 229 errors with strict

File: Makefile (hunk 1 above)
Comment (fspot) [reply Makefile:23]: I find reassuring to know that my tests respect the type annotations, + in some situations it can help to spot wrong type annotations.

**Review by PrettyWood — APPROVED** [review]
